### ISSUE TITLE
Android download manager, Status Code = 16 error

### ISSUE THREAD

**Issue by Noitidart** [wkh237/react-native-fetch-blob#604]
Android download manager is failing to download, even though download succeeds without download manager.

* please provide the version of installed library and RN project.
  * "react-native-fetch-blob": "^0.10.8",
* a sample code snippet/repository is very helpful to spotting the problem.
  ```
        try {
            const res = await RNFetchBlob.config({
                addAndroidDownloads: {
                    useDownloadManager: true,
                    mime: 'image/png',
                    description: 'hiiii'
                }
            })
            .fetch('GET', 'https://myseti.net/api/files/25844492-f7b7-4dcb-a9f7-4d65678ce6c7?version=original');
            console.log('res:', res);
        } catch(ex) {
            console.warn('ex when downloading, ex:', ex);
        }
  ```

The above gives error:

> "Error: Download manager failed to download from  https://myseti.net/api/files/25844492-f7b7-4dcb-a9f7-4d65678ce6c7?version=original. Statu Code = 16"

However this code works fine:

```
            const res = await RNFetchBlob.config({
                fileCache: true,
                appendExt: 'png'
            })
            .fetch('GET', 'https://myseti.net/api/files/25844492-f7b7-4dcb-a9f7-4d65678ce6c7?version=original');
```

But this does not use download manager. :(

**Comment by Noitidart** [comment]
Is it possible the cookies from the regular `fetch` call is not being used by the download manager? Is there a way to make it use the cookies?

**Comment by Noitidart** [comment]
Here are the notifications that show after the above fails:

![](https://screenshots.firefoxusercontent.com/images/118b58db-4420-426e-a6ff-4b2219d2f95d.png)

That one on the bottom with custom notification title is because i used the `.config` options to set a custom notif title.

**Comment by Noitidart** [comment]
Yes this was the problem. I had to do a dummy request, then set `Cookie` header in order for it to work:


        // dummy request to figure out cookie
        const resCookie = await fetchApi('session');
        if (resCookie.status !== 200) return alert('Failed to download, you must not be logged in anymore.');
        const cookie = resCookie.headers.map['set-cookie'][0];

        // download it
        const res = await RNFetchBlob.config({
            fileCache: true,
            addAndroidDownloads: {
                useDownloadManager: true,
                mime: mimeType,
                description
            }
        })
        .fetch('GET', originalUrl, { 'Cookie':cookie });

**Comment by danna88** [comment]
any progress？
